client/dashboard/sites/settings-site-visibility/privacy-form.tsx

```diff
@@ -3,60 +3,118 @@ import {
-			},
-			{
-				label: __( 'Public' ),
-				value: 'public',
-				description: __( 'Your site is visible to everyone.' ),
-			},
-			{
-				label: __( 'Private' ),
-				value: 'private',
-				description: __(
-					'Your site is only visible to you and logged-in members you approve. Everyone else will see a log in screen.'
-				),
-			},
-		],
-	},
-];
-
-const form = {
-	type: 'regular' as const,
-	fields: [ { id: 'wpcom_site_visibility', labelPosition: 'none' } as SimpleFormField ],
-};
-
 export function PrivacyForm( {
+	site,
 	settings,
 	mutation,
 }: {
+	site: Site;
 	settings: SiteSettings;
 	mutation: UseMutationResult< Partial< SiteSettings >, Error, Partial< SiteSettings >, unknown >;
 } ) {
 	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
 	const [ formData, setFormData ] = useState( {
 		wpcom_site_visibility: settings.wpcom_site_visibility,
+		wpcom_discourage_search_engines: settings.wpcom_discourage_search_engines,
+		wpcom_prevent_third_party_sharing:
+			settings.wpcom_discourage_search_engines || settings.wpcom_prevent_third_party_sharing,
 	} );
 
+	const fields: Field< SiteSettings >[] = useMemo(
+		() => [
+			{
+				id: 'wpcom_site_visibility',
+				Edit: 'toggleGroup',
+				elements: [
+					{
+						label: __( 'Coming soon' ),
+						value: 'coming-soon',
+						description: __(
+							'Your site is hidden from visitors behind a "Coming Soon" notice until it is ready for viewing.'
+						),
+					},
+					{
+						label: __( 'Public' ),
+						value: 'public',
+						description: __( 'Your site is visible to everyone.' ),
+					},
+					{
+						label: __( 'Private' ),
+						value: 'private',
+						description: __(
+							'Your site is only visible to you and logged-in members you approve. Everyone else will see a log in screen.'
+						),
+					},
+				],
+			},
+			{
+				id: 'wpcom_discourage_search_engines',
+				Edit: ( { field, onChange, data, hideLabelFromVision } ) => (
+					<CheckboxControl
+						__nextHasNoMarginBottom
+						label={ hideLabelFromVision ? '' : field.label }
+						checked={ field.getValue( { item: data } ) }
+						onChange={ () => {
+							onChange( { [ field.id ]: ! field.getValue( { item: data } ) } );
+						} }
+						help={ field.description }
+					/>
+				),
+				label: __( 'Discourage search engines from indexing this site' ),
+				description: __(
+					'This does not block access to your site — it is up to search engines to honor your request.'
+				),
+			},
+			{
+				id: 'wpcom_prevent_third_party_sharing',
+				Edit: ( { field, onChange, data, hideLabelFromVision } ) => (
+					<CheckboxControl
+						__nextHasNoMarginBottom
+						label={ hideLabelFromVision ? '' : field.label }
+						checked={ field.getValue( { item: data } ) }
+						disabled={ data.wpcom_discourage_search_engines }
+						onChange={ () => {
+							onChange( { [ field.id ]: ! field.getValue( { item: data } ) } );
+						} }
+						help={ createInterpolateElement(
```

Comment:
Ideally I'd be able to use the field `description` for this text. However `description` is annotated as a `string`, even though `BaseControlProps.help` can accept a `ReactNode`.